windows-agent/internal/proservices/landscape/executor.go

```diff
@@ -189,3 +196,119 @@ func (e executor) setDefault(ctx context.Context, cmd *landscapeapi.Command_SetD
 func (e executor) shutdownHost(ctx context.Context, cmd *landscapeapi.Command_ShutdownHost) error {
 	return gowsl.Shutdown(ctx)
 }
+
+func installFromMicrosoftStore(ctx context.Context, distro gowsl.Distro) error {
```

Comment:
I would defer decore.OnError so that we prewrap with `can’t install from Microsoft Store:`